packages/core/src/odata/v4/extract-odata-etag.ts

```diff
@@ -8,5 +8,5 @@
 export function extractODataEtagV4(
   json: Record<string, any>
 ): string | undefined {
-  return '@odata.etag' in json ? json['@odata.etag'] : undefined;
+  return json ? json['@odata.etag'] : undefined;
```

Comment:
```suggestion
  return json.?['@odata.etag'];
```

Reply:
should be `return json?.['@odata.etag'];`